src/app/(marketing)/create-profile/_components/StartProfile.tsx

```diff
@@ -9,119 +11,211 @@ import {
-import { AvatarFallback } from "@radix-ui/react-avatar"
-import { useState } from "react"
+import useUser from "components/[guild]/hooks/useUser"
+import useDropzone from "hooks/useDropzone"
+import usePinata from "hooks/usePinata"
+import { useEffect, useState } from "react"
 import { FormProvider, useForm } from "react-hook-form"
 import { z } from "zod"
 import FarcasterImage from "/src/static/socialIcons/farcaster.svg"
+import { useCreateProfile } from "../_hooks/useCreateProfile"
+import { profileSchema } from "../schemas"
 import { OnboardingChain } from "../types"
 
-const formSchema = z.object({
-  name: z.string().max(100, { message: "Name cannot exceed 100 characters" }),
-  username: z
-    .string()
-    .min(1, { message: "Handle is required" })
-    .max(100, { message: "Handle cannot exceed 100 characters" })
-    .superRefine((value, ctx) => {
-      const pattern = /^[\w\-.]+$/
-      const isValid = pattern.test(value)
-      if (!isValid) {
-        ctx.addIssue({
-          code: z.ZodIssueCode.custom,
-          message:
-            "Handle must only contain either alphanumeric, hyphen, underscore or dot characters",
-        })
-      }
-    }),
-})
+enum CreateMethod {
+  FillByFarcaster,
+  FromBlank,
+}
 
-// TODO: use ConnectFarcasterButton
 export const StartProfile: OnboardingChain = () => {
-  const form = useForm<z.infer<typeof formSchema>>({
-    resolver: zodResolver(formSchema),
+  const { farcasterProfiles = [], isLoading: isUserLoading } = useUser()
+  const farcasterProfile = farcasterProfiles.at(0)
+  const [method, setMethod] = useState<CreateMethod>()
+  const { toast } = useToast()
+
+  useEffect(() => {
+    if (!farcasterProfile || method !== CreateMethod.FillByFarcaster) return
+    form.setValue(
+      "name",
+      farcasterProfile.username ?? form.getValues()?.name ?? "",
+      { shouldValidate: true }
+    )
+    form.setValue("profileImageUrl", farcasterProfile.avatar, {
+      shouldValidate: true,
+    })
+  }, [farcasterProfile, method])
+
+  const form = useForm<z.infer<typeof profileSchema>>({
+    resolver: zodResolver(profileSchema),
     defaultValues: {
       name: "",
       username: "",
     },
     mode: "onTouched",
   })
 
-  function onSubmit(values: z.infer<typeof formSchema>) {
-    console.log(values)
+  const createProfile = useCreateProfile()
+  async function onSubmit(values: z.infer<typeof profileSchema>) {
+    createProfile.onSubmit(values)
+    console.log("onSubmit", values)
+  }
+
+  const { isUploading, onUpload } = usePinata({
+    control: form.control,
+    fieldToSetOnSuccess: "profileImageUrl",
+    onError: (error) => {
+      toast({
+        variant: "error",
+        title: "Failed to upload file",
+        description: error,
+      })
+    },
+  })
+
+  const [uploadProgress, setUploadProgress] = useState(0)
+  const { isDragActive, fileRejections, getRootProps } = useDropzone({
+    multiple: false,
+    noClick: false,
+    maxSizeMb: 4,
```

Comment:
Let's just leave it on the default file size (afaik it's 5MB, but it'll fallback to that value if you don't specify the `maxSizeMb` prop). I'm pretty sure we'll get a feedback to increase it later. :D